draco/analysis/powerspec.py

```diff
@@ -0,0 +1,1504 @@
+                    uv_mask, u, v, u_min_lambda, u_max_lambda
+                )
+
+                w_flat, _, _ = reshape_data_cube(W, u, v, u_min_lambda, u_max_lambda)
+
+                ps3D_flat.append(ps_flat[m_flat])
+                weight_flat.append(w_flat[m_flat])
+
+            # Apply the mask here to the [u,v]
+            uu_flat = uu_flat[m_flat]
+            vv_flat = vv_flat[m_flat]
+
+            # Reshape the array of shape [ndelay,nvis]
+            ps3D_flat = np.array(ps3D_flat).reshape(kpar.size, -1)
+            weight_flat = np.array(weight_flat).reshape(kpar.size, -1)
+
+            # convert to Fourier modes and estimate kperp
+            ku = u_to_kperp(
+                uu_flat, redshift, ps.cosmology
+            )  # convert the u-array to fourier modes
+            kv = u_to_kperp(
+                vv_flat, redshift, ps.cosmology
+            )  # convert the v-array to fourier modes
+            kperp = np.sqrt(ku**2 + kv**2)
+
+            # Generate the signal window mask corresponding to the delay cut
+            signal_mask = np.ones_like(ps3D_flat, dtype=bool)
+
+            if self.delay_cut > 0.0:
+                kpar_lim = delays_to_kpara(self.delay_cut, redshift, ps.cosmology)
+                ibins = np.where((kpar > -kpar_lim) & (kpar < kpar_lim))
+                signal_mask[ibins, :] = False
+
+            # Estimate the 1D power spectrum
+            (
+                pspec_1D.k1D[pp],
+                pspec_1D.ps1D[pp],
+                pspec_1D.samp_var[pp],
+                pspec_1D.var[pp],
+            ) = get_1d_ps(
+                ps3D_flat,
+                kperp,
+                kpar,
+                signal_window=signal_mask,
+                Nbins_3D=self.Nbins_3D,
+                weight_cube=weight_flat
+                ** 2,  # square the weight for inverse variance weighted avg of power spectrum
+                logbins_3D=self.logbins_3D,
+            )
+        return pspec_1D
+
+
+def f2z(freq):
+    """Convert frequency to redshift for the 21 cm line.
+
+    Parameters
+    ----------
+    freq : float
+      frequency in MHz
+
+    Returns
+    -------
+    redshift: float
+    """
+    return units.nu21 / freq - 1
+
+
+def z2f(z):
+    """Convert redshift to frequency for the 21 cm line.
+
+    Parameters
+    ----------
+    z : float
+     redshift
+
+    Returns
+    -------
+    frequency: float
+       frequency in MHz
+    """
+    return units.nu21 / (z + 1)
+
+
+def dRperp_dtheta(z, cosmo=None):
+    """Conversion factor from angular size (radian) to transverse comoving distance (Mpc/h) at a specific redshift: [h^-1 Mpc / radians].
+
+    Parameters
+    ----------
+    z : float
+     redshift
+    cosmo: Cosmology object
+        Default is cora.util.Cosmology() default.
+
+    Returns
+    -------
+    comoving transverse distance: float. unit in [h^-1 Mpc].
+    """
+    if cosmo is None:
+        cosmo = get_cosmo()
+    return cosmo.comoving_distance(z)
```

Comment:
```suggestion
    return cosmo.proper_distance(z)
```
I think this is what you want here, although they should give the same result of Omega_K = 0.